GameServerLib/Content/NavGrid.cs

```diff
@@ -77,8 +83,8 @@ public List<Vector2> GetPath(Vector2 from, Vector2 to)
                     {
                         // if the neighbor in the closed list - skip
                         if (closedList.TryGetValue(ncell.Id, out tempCell)) continue;
-                        // not walkable - skip
-                        if (ncell.HasFlag(this, NavigationGridCellFlags.NOT_PASSABLE) || ncell.HasFlag(this, NavigationGridCellFlags.SEE_THROUGH))
+                        // not walkable or not sightable or occupied by static object - skip
+                        if (ncell.HasFlag(this, NavigationGridCellFlags.NOT_PASSABLE) || ncell.HasFlag(this, NavigationGridCellFlags.SEE_THROUGH) || _colHandler.IsOccupiedByStaticObject(TranslateFromNavGrid(new Vector2(ncell.X, ncell.Y)), colRadius))
```

Comment:
You're checking every cell if it has a static object specifically while you could also flag the cell for having a static object. This is very inefficient

Reply:
This is the layer solution. The problem is with updating the static (like Anivia wall).